task_router_workflow_test.go

```diff
@@ -0,0 +1,62 @@
+package twilio
+
+import (
+	"context"
+	"net/url"
+	"testing"
+
+	"github.com/kevinburke/twilio-go/testdata"
+)
+
+func TestGetWorkflow(t *testing.T) {
+	t.Parallel()
+	client, server := getServer(testdata.WorkflowResponse)
+	defer server.Close()
+
+	sid := "WW63868a235fc1cf3987e6a2b67346273f"
+	taskReservationTimeout := 120
+
+	workflow, err := client.TaskRouter.Workspace("WS58f1e8f2b1c6b88ca90a012a4be0c279").Workflows.Get(context.Background(), sid)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if workflow.Sid != sid {
+		t.Errorf("task router workflow: got sid %q, want %q", workflow.Sid, sid)
+	}
+
+	if workflow.TaskReservationTimeout != taskReservationTimeout {
+		t.Errorf("task router workflow: got sid %q, want %q", workflow.TaskReservationTimeout, taskReservationTimeout)
+	}
+}
+
+func TestCreateWorkflow(t *testing.T) {
+	t.Parallel()
+	client, server := getServer(testdata.WorkflowCreateResponse)
+	defer server.Close()
+
+	data := url.Values{}
+	newname := "Sales, Marketing, Support Workflow"
+	data.Set("FriendlyName", newname)
+	workspaceSid := "WS7a2aa7d8acc191786ad3c647c5fc3110"
+
+	workflow, err := client.TaskRouter.Workspace(workspaceSid).Workflows.Create(context.Background(), data)
```

Comment:
Getting an error of `json: cannot unmarshal object into Go struct field Workflow.configuration of type string`

Guess it's because `configuration` is a json object, any ideas how to parse correctly?

Reply:
This is a problem - in the Workflow Create response it's

```
"configuration": { ... }
```

In the get response it's:

```
"configuration": "..."
```

It can't be both. This seems like an error in the API.

Reply:
Going to try to see what the actual API is returning right now